src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -2953,6 +2976,566 @@ ImmutableArray<IArgumentOperation> makeArguments(Lazy<ImmutableArray<IArgumentOp
+
+            void initializeLoop()
+            {
+                if (isObjectLoop)
+                {
+                    // PROTOTYPE(dataflow): Handle loops over object
+                    throw new NotImplementedException();
+                }
+                else
+                {
+                    int captureId = _availableCaptureId++;
+                    IOperation controlVarReferenceForInitialization = getLoopControlVariableRefernce(forceImplicit: false, captureId);
+                    CaptureResultIfNotAlready(controlVarReferenceForInitialization.Syntax, captureId, controlVarReferenceForInitialization);
+
+                    int initialValueId = VisitAndCapture(operation.InitialValue);
+                    limitValueId = VisitAndCapture(operation.LimitValue);
+                    stepValueId = VisitAndCapture(operation.StepValue);
+
+                    if (userDefinedInfo != null)
+                    {
+                        // PROTOTYPE(dataflow): Handle "user-defined" loops
+                        throw new NotImplementedException();
+
+                        //        ' calculate and cache result of a positive check := step >= (step - step).
+                        //AddPlaceholderReplacement(forStatement.OperatorsOpt.LeftOperandPlaceholder, rewrittenStep)
+                        //AddPlaceholderReplacement(forStatement.OperatorsOpt.RightOperandPlaceholder, rewrittenStep)
+
+                        //Dim subtraction = VisitExpressionNode(forStatement.OperatorsOpt.Subtraction)
+
+                        //UpdatePlaceholderReplacement(forStatement.OperatorsOpt.RightOperandPlaceholder, subtraction)
+
+                        //Dim greaterThanOrEqual = VisitExpressionNode(forStatement.OperatorsOpt.GreaterThanOrEqual)
+
+                        //positiveFlag = New SynthesizedLocal(_currentMethodOrLambda, greaterThanOrEqual.Type, SynthesizedLocalKind.ForDirection, blockSyntax)
+                        //locals.Add(positiveFlag)
+
+                        //cacheAssignments.Add(New BoundAssignmentOperator(forStatement.OperatorsOpt.Syntax,
+                        //                                                 New BoundLocal(forStatement.OperatorsOpt.Syntax,
+                        //                                                                positiveFlag,
+                        //                                                                positiveFlag.Type),
+                        //                                                 greaterThanOrEqual,
+                        //                                                 suppressObjectClone:= True, type:= positiveFlag.Type))
+
+                        //RemovePlaceholderReplacement(forStatement.OperatorsOpt.LeftOperandPlaceholder)
+                        //RemovePlaceholderReplacement(forStatement.OperatorsOpt.RightOperandPlaceholder)
+                    }
+                    else if (!operation.StepValue.ConstantValue.HasValue &&
```

Comment:
Can `StepValue` be null? If not, we probably want to update the [doc comment](http://source.roslyn.io/#Microsoft.CodeAnalysis/Operations/Loops/IForToLoopOperation.cs,36).